Standards/scs-0112-v1-volume-type-standard.md

```diff
@@ -0,0 +1,227 @@
+
+## Design Considerations
+
+All Considerations can be looked up in detail in the [Decision Record for the Volume Type Standard.](https://github.com/SovereignCloudStack/standards/blob/main/Standards/scs-0111-v1-volume-type-decisions.md)
+
+To test whether a deployment has volume types with certain aspects, the discoverability of the parameters in the volume type has to be given. The following table shows, which aspects are considered in this standard. Furthermore the table shows, according to the current upstream implementation, whether the feature is discoverable, and who can discover this. The comments state what can be done to help with the discoverability and other important restrictions. The last column says how the standard is handled, which is currently through the addition of a standardized phrase at the beginning of the description of a volume type.
+
+| Aspect | Part of Standard | Discoverability | comments | concluded standardized description |
+| ---- | ---- | ---- | ---- | ------ |
+| encryption | **Recommended** | admin only | Upstream work for extra_spec: encrypted=True/False needed | **"[encrypted]"** |
+| Replication | **Recommended** | backend-dependend | Upstream: setable extra_spec would help | **"[replicated]"** |
+| AZs | **Optional** | - | Upstream: setable extra_spec would help | - |
+| multiattach | **Optional** | yes | - | - |
+| Number of Replicas, etc | **Optional** | backend-dependend | Upstream: setable extra_spec would help | - |
+| Volume QoS | **Optional** | admin only | depends on the admin only volume qos object | - |
+| Backend name | no | - | - | - |
+
+In addition it is possible to use multiple of those aspects within one volume type. SCS will only ever look, if there is a volume type that has an aspect, but they don't have to be different volume types.
+Example: one volume type that uses LUKS-encryption with a ceph storage with inherent replication would fulfill all recommendations of this standard.
+
+The current downside is that users without admin rights currently cannot see the recommended aspects. We want and need to change this in the OpenStack workflow through adding discoverable properties (e.g. "encrypted"="true" and "replicated"="true") that are also visible for users. This way we will be able to automatically check whether a volume type with encyrption and or replication is present. It should look like this example for encryption:
```

Comment:
I would not use first person plural. And I would not write, that SCS will extend OpenStack. I would just point out, that an extension in OpenStack is required, to support/fulfill this standard.